x-pack/plugins/snapshot_restore/public/app/sections/home/snapshot_list/snapshot_list.tsx

```diff
@@ -4,8 +4,135 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
+import React, { Fragment } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
 
-export const SnapshotList: React.FunctionComponent = () => {
-  return <div>List of snapshots</div>;
+import { SectionError, SectionLoading } from '../../../components';
+import { BASE_PATH } from '../../../constants';
+import { useAppDependencies } from '../../../index';
+import { documentationLinksService } from '../../../services/documentation';
+import { loadSnapshots } from '../../../services/http';
+
+import { SnapshotDetails } from './snapshot_details';
+import { SnapshotTable } from './snapshot_table';
+
+import { EuiButton, EuiEmptyPrompt } from '@elastic/eui';
+
+interface MatchParams {
+  repositoryName?: string;
+  snapshotId?: string;
+}
+
+interface Props extends RouteComponentProps<MatchParams> {}
+
+export const SnapshotList: React.FunctionComponent<Props> = ({
+  match: {
+    params: { repositoryName: pathRepositoryName, snapshotId: pathSnapshotId },
+  },
+  history,
+}) => {
+  const {
+    core: {
+      i18n: { FormattedMessage },
+    },
+  } = useAppDependencies();
+
+  const {
+    error,
+    loading,
+    data: { snapshots },
+    request: reload,
+  } = loadSnapshots();
+
+  const openSnapshotDetails = (repositoryName: string, snapshotId: string) => {
+    history.push(`${BASE_PATH}/snapshots/${repositoryName}/${snapshotId}`);
```

Comment:
@jen-huang I was able to remove the `useState` and `useEffect` hooks and just depend upon the route params. I think we can do the same thing for the RepositoryList.